tests/robustness/snapshotter.go

```diff
@@ -3,13 +3,27 @@
 // can be used to test other environments.
 package robustness
 
+import (
+	"io"
+	"time"
+)
+
 // Snapshotter is an interface that describes methods
 // for taking, restoring, deleting snapshots, and
 // tracking them by a string snapshot ID.
 type Snapshotter interface {
-	CreateSnapshot(sourceDir string, opts map[string]string) (snapID string, err error)
-	RestoreSnapshot(snapID string, restoreDir string, opts map[string]string) error
 	DeleteSnapshot(snapID string, opts map[string]string) error
 	RunGC(opts map[string]string) error
 	ListSnapshots() ([]string, error)
+
+	CreateSnapshot(sourceDir string, opts map[string]string) (snapID string, fingerprint []byte, stats *CreateSnapshotStats, err error)
+	RestoreSnapshot(snapID, restoreDir string, opts map[string]string) ([]byte, error)
+	RestoreSnapshotCompare(snapID, restoreDir string, validationData []byte, reportOut io.Writer, opts map[string]string) error
```

Comment:
Would be great if all these could take a `context.Context` argument that could be passed in via the test method facing engine methods.

Reply:
Change made here (branched off this PR) https://github.com/kastenhq/kopia/pull/129